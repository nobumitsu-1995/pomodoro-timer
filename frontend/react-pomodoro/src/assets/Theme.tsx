import React from 'react'
import { ThemeProvider } from 'styled-components'

type Props = {
  children: React.ReactNode
}

const Theme: React.FC<Props> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const theme = {
  color: {
    black: '#000',
    white: '#fff',
    red: 'red',
    blue: 'blue',
    green: 'green',
  },
  size: {
    small: '1.0rem',
    normal: '1.2rem',
    big: '2.0rem',
  },
} as const

export default Theme
