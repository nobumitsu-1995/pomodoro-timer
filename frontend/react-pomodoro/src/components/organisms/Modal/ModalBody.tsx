import React from 'react'
import styled, { keyframes } from 'styled-components'
import { useModalContext } from '../../../lib/functions/ModalContext'
import { Paper } from '../../atoms'

const ModalBody = () => {
  const { open, setOpen, contents, setContents } = useModalContext()
  const handleClick = () => {
    setOpen(false)
    setContents('')
  }
  return (
    <StyledDiv open={open} onClick={handleClick}>
      <StyledPaper
        padding='0'
        onClick={(e) => e.stopPropagation()}
        width='80vw'
      >
        {contents}
      </StyledPaper>
    </StyledDiv>
  )
}

export default ModalBody

const fadeIn = keyframes`
  from {
    opacity: 0;
  }
  1% {
    display: block;
  }
  to {
    opacity: 1;
  }
`

const StyledDiv = styled.div<{ open: boolean }>`
  width: 100vw;
  height: 100vh;
  display: none;
  opacity: 0;
  position: fixed;
  top: 0;
  left: 0;
  z-index: 10;
  animation: ${fadeIn} 0.2s linear;
  ${({ open }) =>
    open &&
    `
      display: block;
      opacity: 1
    `}
`

const StyledPaper = styled(Paper)`
  position: fixed;
  top: 0;
  bottom: 0;
  right: 0px;
  left: 0px;
  z-index: 15;
  max-height: 80vh;
  margin: auto;
  box-shadow: 40px 40px 80px #999999, -40px -40px 80px #ffffff;
  border-radius: 10px;
  overflow: hidden;
  overflow-y: scroll;
`
