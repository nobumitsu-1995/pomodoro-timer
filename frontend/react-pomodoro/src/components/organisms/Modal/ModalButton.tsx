import React from 'react'
import { useModalContext } from '../../../lib/functions/ModalContext'
import { Button } from '../../atoms'

export type Props = {
  children: React.ReactNode | string
  modalContent: React.ReactNode | string
  borderradius?: string
  size: string
}

const ModalButton: React.FC<Props> = ({
  children,
  modalContent,
  borderradius,
  size,
}) => {
  const { open, setOpen, setContents } = useModalContext()
  const handleClick = () => {
    setOpen(!open)
    setContents(modalContent)
  }
  return (
    <Button size={size} borderradius={borderradius} onClick={handleClick}>
      {children}
    </Button>
  )
}

export default ModalButton
