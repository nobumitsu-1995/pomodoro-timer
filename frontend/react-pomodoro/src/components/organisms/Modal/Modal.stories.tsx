import React from 'react'
import { Meta, Story } from '@storybook/react'
import { within, userEvent } from '@storybook/testing-library'
import ModalButton, { Props } from './ModalButton'
import { ModalProvider } from '../../../lib/functions/ModalContext'
import ModalBody from './ModalBody'

export default {
  title: 'organisms/Modal',
  component: ModalButton,
  argTypes: {},
  decorators: [
    (Story) => (
      <ModalProvider>
        <ModalBody />
        <div style={{ maxWidth: '320px', margin: 'auto' }}>
          <Story />
        </div>
      </ModalProvider>
    ),
  ],
} as Meta

const Template: Story<Props> = (args) => <ModalButton {...args} />

export const Default = Template.bind({})
Default.args = {
  children: 'Button',
  modalContent: (
    <>
      <p>modal</p>
      <p>modal</p>
      <p>modal</p>
      <p>modal</p>
      <p>modal</p>
      <p>modal</p>
      <p>modal</p>
    </>
  ),
  borderradius: '20px',
}

export const isOpen = Template.bind({})
isOpen.args = {
  ...Default.args,
}
isOpen.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement)
  await userEvent.click(canvas.getByText('Button'))
}
