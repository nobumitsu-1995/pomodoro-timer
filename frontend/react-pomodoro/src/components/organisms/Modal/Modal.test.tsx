/* eslint-disable react/react-in-jsx-scope */
import * as stories from './Modal.stories'
import { composeStories } from '@storybook/react'
import { render } from '@testing-library/react'
import Theme from 'src/assets/styles/Theme'
import ModalBody from './ModalBody'

describe('snapshot', () => {
  Object.entries(composeStories(stories)).forEach(([name, Component]) => {
    describe(name, () => {
      it('should render correctly', () => {
        const tree = render(
          <Theme>
            <Component {...Component.args} {...Component.play} />
          </Theme>
        ).asFragment()
        expect(tree).toMatchSnapshot()
      })
    })
  })

  it('Modal Body', () => {
    const tree = render(
      <Theme>
        <ModalBody />
      </Theme>
    ).asFragment()
    expect(tree).toMatchSnapshot()
  })
})
