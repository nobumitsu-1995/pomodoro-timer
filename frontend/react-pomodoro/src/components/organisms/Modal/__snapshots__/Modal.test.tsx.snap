// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Default should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-hKMtZM iHZNaF"
  >
    <div
      class="Paper__StyledDiv-sc-eiawt3-0 Paper___StyledStyledDiv-sc-eiawt3-1 mEzeE bFEtGg sc-eCYdqJ girqUe"
    />
  </div>
  <div
    style="max-width: 320px; margin: auto;"
  >
    <button
      class="Button__StyledButton-sc-mes1hd-0 Button___StyledStyledButton-sc-mes1hd-1 fxBFHR hvHmUL"
      type="button"
    >
      Button
    </button>
  </div>
</DocumentFragment>
`;

exports[`snapshot Modal Body 1`] = `
<DocumentFragment>
  <div
    class="sc-hKMtZM iHZNaF"
  >
    <div
      class="Paper__StyledDiv-sc-eiawt3-0 Paper___StyledStyledDiv-sc-eiawt3-1 mEzeE bFEtGg sc-eCYdqJ girqUe"
    />
  </div>
</DocumentFragment>
`;

exports[`snapshot isOpen should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-hKMtZM iHZNaF"
  >
    <div
      class="Paper__StyledDiv-sc-eiawt3-0 Paper___StyledStyledDiv-sc-eiawt3-1 mEzeE bFEtGg sc-eCYdqJ girqUe"
    />
  </div>
  <div
    style="max-width: 320px; margin: auto;"
  >
    <button
      class="Button__StyledButton-sc-mes1hd-0 Button___StyledStyledButton-sc-mes1hd-1 fxBFHR hvHmUL"
      type="button"
    >
      Button
    </button>
  </div>
</DocumentFragment>
`;
